lib/activerecord-multi-tenant/model_extensions.rb

```diff
@@ -1,39 +1,45 @@
 module MultiTenant
   module ModelExtensionsClassMethods
-    def multi_tenant(tenant, options = {})
-      # Workaround for https://github.com/citusdata/citus/issues/687
-      if to_s.underscore.to_sym == tenant
-        before_create -> { self.id ||= self.class.connection.select_value("SELECT nextval('" + [self.class.table_name, self.class.primary_key, 'seq'].join('_') + "'::regclass)") }
-      end
-
-      # Typically we don't need to run on the tenant model itself
-      if to_s.underscore.to_sym != tenant
-        MultiTenant.set_tenant_klass(tenant)
+    DEFAULT_ID_FIELD = 'id'.freeze
 
+    def multi_tenant(tenant_name, options = {})
+      if to_s.underscore.to_sym == tenant_name
+        # This is the tenant model itself. Workaround for https://github.com/citusdata/citus/issues/687
+        before_create -> { self.id ||= self.class.connection.select_value("SELECT nextval('" + [self.class.table_name, self.class.primary_key, 'seq'].join('_') + "'::regclass)") }
+      else
         class << self
           def scoped_by_tenant?
             true
           end
 
+          # Allow partition_key to be set from a superclass if not already set in this class
           def partition_key
-            @partition_key
+            @partition_key ||= ancestors.detect{ |k| k.instance_variable_get(:@partition_key) }
```

Comment:
I tried an alternative method, using const_set instead of traversing ancestors but it failed in some rubies / rails versions when running appraisal... since this worked I didn't go too deep